gamestonk_terminal/jupyter/dashboards/shortdata.ipynb

```diff
@@ -0,0 +1,213 @@
+    "        today = datetime.now().date()\n",
+    "        idx = 0\n",
+    "        len_df = 0\n",
+    "        while len_df < self.days_slider.value:\n",
+    "            date = today - timedelta(days = idx)\n",
+    "            r = requests.get(\n",
+    "                f\"https://cdn.finra.org/equity/regsho/daily/CNMSshvol{date.strftime('%Y%m%d')}.txt\"\n",
+    "            )\n",
+    "            if r.status_code == 200:\n",
+    "                self.df = pd.concat([self.df,pd.read_csv(io.StringIO(r.text), sep = \"|\")], axis=0)\n",
+    "                len_df += 1\n",
+    "            idx +=1\n",
+    "        self.df = self.df[self.df.Date > 20100101]\n",
+    "        self.df.Date = self.df[\"Date\"].apply(lambda x: datetime.strptime(str(x), \"%Y%m%d\"))\n",
+    "\n",
+    "\n",
+    "    def update(self):\n",
+    "        if not self.df.empty:\n",
+    "            temp = self.df.groupby(\"Symbol\")[[\"ShortVolume\",\"TotalVolume\"]].agg(\"sum\").sort_values(by=\"ShortVolume\",ascending=False).head(self.count_slider.value)[::-1]\n",
+    "            self.fig,self.ax = plt.subplots(figsize = (6,6))\n",
+    "            self.ax.barh(temp.index, temp.TotalVolume, alpha = 0.4, label = \"Total Volume\")\n",
+    "            self.ax.barh(temp.index,temp.ShortVolume, label = \"Short Volume\")\n",
+    "            self.ax.set_title(f\"Top {self.count_slider.value} Short Volume in Last {self.days_slider.value} Days\")\n",
+    "            self.ax.legend()\n",
+    "            self.fig.tight_layout()\n",
+    "            plt.show()\n",
+    "\n",
+    "    def ticker_plot(self):\n",
+    "        stock_data = self.df.copy().loc[self.df.Symbol == self.stock_input.value,[\"Date\",\"ShortVolume\",\"TotalVolume\"]]\n",
+    "        self.fig2,self.ax2 = plt.subplots(figsize = (6,6))\n",
+    "        self.ax2.plot(stock_data.Date, stock_data.TotalVolume, alpha = 0.4, label = \"Total Volume\")\n",
+    "        self.ax2.plot(stock_data.Date,stock_data.ShortVolume, label = \"Short Volume\")\n",
+    "        self.ax2.set_title(f\"Stock Volume and Short Volume for {self.stock_input.value.upper()}\")\n",
+    "        self.ax2.legend()\n",
+    "        self.fig2.autofmt_xdate()\n",
+    "        self.fig2.tight_layout()\n",
+    "        plt.show()\n",
+    "\n",
+    "    def build_app(self):\n",
+    "        title_html = \"\"\"\n",
+    "<h2>Finra Short Data</h2>\n",
+    "<p>This widget downloads the consolidated NMS short data from FINRA and aggregates the data by summing over the entire time period.</p>\n",
+    "<p>Note that clicking the 'Load Data' button will reload all data.  This can get time consuming, so if you pick a few hundred days, expect a few minutes for loading time.</p>\n",
+    "\"\"\"\n",
+    "        middle_html = \"\"\"\n",
+    "Here we allow the user to query for a single stock.  This will work with the loaded data.  Note that if you want to reload the data, this will once again tke some time.\n",
```

Comment:
Spelling error.

Reply:
You shouldn't request changes on things like this lol.  Just comment on the line.